### PR TITLE
Add profile stats to the CPU profiler

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Work towards https://github.com/flutter/devtools/issues/3861

Screenshot 2023-03-01 at 4 19 54 PM
Screenshot 2023-03-01 at 4 19 41 PM

While I was here, I also created a new file `profiler_controls.dart` and pulled some controls from cpu_profiler.dart out into the new file. This was to make the cpu_profiler.dart file more manageable.

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart [hunk 1]
```diff
@@ -365,224 +367,62 @@ class _CpuProfilerState extends State<CpuProfiler>
   }
 }
 
-class DisplayTreeGuidelinesToggle extends StatelessWidget {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart:368]: all the deletes over here were moved to profiler_controls.dart with no functional changes. git diff picked this up weird unfortunately.

File: packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart [hunk 2]
```diff
@@ -365,224 +367,62 @@ class _CpuProfilerState extends State<CpuProfiler>
...
-  const DisplayTreeGuidelinesToggle();
+// TODO(kenz): one improvement we could make on this is to show the denominator
+// for filtered profiles (e.g. 'Sample count: 10/14), or to at least show the
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart:371]: review all the additions from here to the bottom of the file as new code - not a modification of code on the left.

File: packages/devtools_app/lib/src/screens/profiler/panes/controls/profiler_controls.dart [hunk 3]
```diff
@@ -0,0 +1,241 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/profiler/panes/controls/profiler_controls.dart:1]: no functional changes to this file. all moves from cpu_profiler.dart

File: packages/devtools_app/test/cpu_profiler/cpu_profiler_test.dart [hunk 4]
```diff
@@ -148,6 +151,7 @@ void main() {
         expect(find.byType(TabBar), findsOneWidget);
         expect(find.byKey(CpuProfiler.dataProcessingKey), findsNothing);
         expect(find.byType(CpuBottomUpTable), findsOneWidget);
+        expect(find.byType(CpuProfileStats), findsOneWidget);
```
Comment (bkonyi) [inline packages/devtools_app/test/cpu_profiler/cpu_profiler_test.dart:154]: Do we want a test that checks the values are rendered properly?

**Review by bkonyi — APPROVED** [review]
LGTM with a couple of comments

**Pull request merged**